sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -423,20 +388,12 @@ public virtual async Task<Response> DeleteConfigurationSettingAsync(Configuratio
         /// <returns>A response indicating the success of the operation.</returns>
         public virtual Response DeleteConfigurationSetting(ConfigurationSetting setting, bool onlyIfUnchanged = false, CancellationToken cancellationToken = default)
         {
-            if (setting == null)
-                throw new ArgumentNullException($"{nameof(setting)}");
-
-            MatchConditions requestOptions = default;
-            if (onlyIfUnchanged)
-            {
-                requestOptions = new MatchConditions();
-                requestOptions.IfMatch = setting.ETag;
-            }
-
+            Argument.AssertNotNull(setting, nameof(setting));
+            MatchConditions requestOptions = onlyIfUnchanged ? new MatchConditions { IfMatch = setting.ETag } : default;
```

Comment:
Do you want to go through and make this change everywhere in the file?  It looked like we were still using the old construction method in some places.

Reply:
Fixed